pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -801,4 +810,12 @@ public void setupOutput(ContextImpl contextImpl) throws Exception {
             Thread.currentThread().setContextClassLoader(this.instanceClassLoader);
         }
     }
+
+    public String getStatsAsString() throws IOException {
+        if (stats != null) {
```

Comment:
We should synchronize this method as well?  

Also line 241 (run() method) accesses the stats variable and the close() method does too. Difference threads access these methods and change the stats variable. It's the same for other variables in this class.